tests/unit/task/test_run.py

```diff
@@ -0,0 +1,54 @@
+from argparse import Namespace
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from dbt.flags import get_flags
+from dbt.task.run import RunTask
+from dbt.tests.util import safe_set_invocation_context
+
+
+@pytest.mark.parametrize(
+    "exception_to_raise, expected_cancel_connections",
+    [
+        (SystemExit, True),
+        (KeyboardInterrupt, True),
+        (Exception, False),
+    ],
+)
+def test_run_task_cancel_connections(exception_to_raise, expected_cancel_connections):
+    safe_set_invocation_context()
+
+    def mock_run_queue(*args, **kwargs):
+        raise exception_to_raise("Test exception")
+
+    with patch.object(RunTask, "run_queue", mock_run_queue), patch.object(
+        RunTask, "_cancel_connections"
+    ) as mock_cancel_connections:
+
+        # TODO clean up this after we have a proper runtime config fixture
```

Comment:
Adjust this before merge

Reply:
https://github.com/dbt-labs/dbt-core/pull/10242 has been merged, so we should be able to clean this up now 👍